handlers/deleteitem/deleteItem_test.go

```diff
@@ -0,0 +1,192 @@
+package deleteitem
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/mount-joy/thelist-lambda/db"
+
+	"github.com/mount-joy/thelist-lambda/data"
+	"github.com/mount-joy/thelist-lambda/handlers/testhelpers"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeleteItemMatch(t *testing.T) {
+	tests := []struct {
+		name        string
+		path        string
+		method      string
+		expectedRes bool
+	}{
+		{
+			name:        "Returns true for a matching path",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "DELETE",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns true for a uppercase ID",
+			path:        "/lists/B6CF642D/items/73BB82C4/",
+			method:      "DELETE",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns true without trailing slash",
+			path:        "/lists/b6cf642d/items/73bb82c4",
+			method:      "DELETE",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns false for list path",
+			path:        "/lists/b6cf642d/",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for list path without trailing slash",
+			path:        "/lists/b6cf642d",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for lists path",
+			path:        "/lists/",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for lists path without trailing slash",
+			path:        "/lists",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for items path",
+			path:        "/lists/b6cf642d/items/",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false when path is empty",
+			path:        "",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a GET request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "GET",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a POST request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a PATCH request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dbMocked := &testhelpers.MockDB{}
+			dbMocked.Test(t)
+			defer dbMocked.AssertExpectations(t)
+
+			input := testhelpers.CreateAPIGatewayV2HTTPRequest(tt.path, tt.method, "")
+			d := deleteItem{db: dbMocked}
+			gotRes := d.Match(input)
+
+			assert.Equal(t, tt.expectedRes, gotRes)
+		})
+	}
+}
+
+func TestDeleteItemHandle(t *testing.T) {
+	tests := []struct {
+		name               string
+		path               string
+		listID             string
+		itemID             string
+		output             *data.Item
+		outputErr          error
+		expectedStatusCode int
+		shouldCallDB       bool
```

Comment:
As in the update PR about maybe using a struct to eliminate the need for this
https://github.com/mount-joy/thelist-lambda/pull/9/files#r516857435